optics-core/src/Optics/Label.hs

```diff
@@ -0,0 +1,201 @@
+--
+-- == Example
+--
+-- Consider the following:
+--
+-- >>> :set -XDataKinds
+-- >>> :set -XFlexibleContexts
+-- >>> :set -XFlexibleInstances
+-- >>> :set -XMultiParamTypeClasses
+-- >>> :set -XOverloadedLabels
+-- >>> :set -XTypeFamilies
+-- >>> :set -XUndecidableInstances
+-- >>> :{
+-- data Human = Human
+--   { humanName :: String
+--   , humanAge  :: Integer
+--   , humanPets :: [Pet]
+--   } deriving Show
+-- data Pet
+--   = Cat  { petName :: String, petAge :: Int, petLazy :: Bool }
+--   | Fish { petName :: String, petAge :: Int }
+--   deriving Show
+-- :}
+--
+-- The following instances can be generated by @makeFieldLabels@ from
+-- @Optics.TH@ in the @optics-th@ package:
+--
+-- >>> :{
+-- instance (a ~ String, b ~ String) => LabelOptic "name" A_Lens Human Human a b where
+--   labelOptic = lensVL $ \f s -> (\v -> s { humanName = v }) <$> f (humanName s)
+-- instance (a ~ Integer, b ~ Integer) => LabelOptic "age" A_Lens Human Human a b where
+--   labelOptic = lensVL $ \f s -> (\v -> s { humanAge = v }) <$> f (humanAge s)
+-- instance (a ~ [Pet], b ~ [Pet]) => LabelOptic "pets" A_Lens Human Human a b where
+--   labelOptic = lensVL $ \f s -> (\v -> s { humanPets = v }) <$> f (humanPets s)
+-- instance (a ~ String, b ~ String) => LabelOptic "name" A_Lens Pet Pet a b where
+--   labelOptic = lensVL $ \f s -> (\v -> s { petName = v }) <$> f (petName s)
+-- instance (a ~ Int, b ~ Int) => LabelOptic "age" A_Lens Pet Pet a b where
+--   labelOptic = lensVL $ \f s -> (\v -> s { petAge = v }) <$> f (petAge s)
+-- instance (a ~ Bool, b ~ Bool) => LabelOptic "lazy" An_AffineTraversal Pet Pet a b where
+--   labelOptic = atraversalVL $ \point f s -> case s of
+--     Cat name age lazy -> (\lazy' -> Cat name age lazy') <$> f lazy
+--     _                 -> point s
+-- :}
+--
+-- Here is some test data:
+--
+-- >>> :{
+-- peter :: Human
+-- peter = Human "Peter" 13 [ Fish "Goldie" 1
+--                          , Cat  "Loopy"  3 False
+--                          , Cat  "Sparky" 2 True
+--                          ]
+-- :}
+--
+-- Now we can ask for Peter's name:
+--
+-- >>> view #name peter
+-- "Peter"
+--
+-- or for names of his pets:
+--
+-- >>> toListOf (#pets % folded % #name) peter
+-- ["Goldie","Loopy","Sparky"]
+--
+-- We can check whether any of his pets is lazy:
+--
+-- >>> orOf (#pets % folded % #lazy) peter
+-- True
+--
+-- or how things might be be a year from now:
+--
+-- >>> peter & over #age (+1) & over (#pets % mapped % #age) (+1)
+-- Human {humanName = "Peter", humanAge = 14, humanPets = [Fish {petName = "Goldie", petAge = 2},Cat {petName = "Loopy", petAge = 4, petLazy = False},Cat {petName = "Sparky", petAge = 3, petLazy = True}]}
+--
+-- Perhaps Peter is going on vacation and needs to leave his pets at home:
+--
+-- >>> peter & set #pets []
+-- Human {humanName = "Peter", humanAge = 13, humanPets = []}
+--
+--
+-- == Structure of 'LabelOptic' instances
+--
+-- /Note:/ You might wonder why instances above are written in form
```

Comment:
`/Note:/` is not needed here as it's the beginning of a new section.

Reply:
Good point, removed. I'm not totally satisfied with the docs here but I don't plan to do any more on them right now.